tests/cedict-service/cedict.test.js

```diff
@@ -0,0 +1,279 @@
+/* eslint-env jest */
+import Cedict from '@lexisCs/cedict-service/cedict'
+import CedictPermanentStorage from '@lexisCs/cedict-service/cedict-permanent-storage.js'
+import CedictConfig from '@lexisCs/configurations/cedict.js'
+import CedictC1 from '@lexisCsCedict/cedict-v20191029-c001.json'
+import CedictC2 from '@lexisCsCedict/cedict-v20191029-c002.json'
+import CedictC3 from '@lexisCsCedict/cedict-v20191029-c003.json'
+import CedictC4 from '@lexisCsCedict/cedict-v20191029-c004.json'
+require('fake-indexeddb/auto')
+
+describe('CedictPermanentStorage class', () => {
+  /**
+   * Replaces an original loader with a stub that will load local test data.
+   *
+   * @param {string} url - A URL of a chunk to be fetched.
+   * @returns {Promise<object>} Returns a promise that is resolved with
+   *          an object containing dictionary data.
+   */
+  Cedict.prototype._loadJson = (url) => {
+    const resources = {
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c001.json': CedictC1,
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c002.json': CedictC2,
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c003.json': CedictC3,
+      'http://data-dev.alpheios.net/cedict/cedict-v20191029-c004.json': CedictC4
+    }
+    return Promise.resolve(resources[url])
+  }
+
+  let dictionary = new Map() // eslint-disable-line prefer-const
+  ;[CedictC1.entries, CedictC2.entries, CedictC3.entries, CedictC4.entries].flat()
```

Comment:
is the ; here a typo?

Reply:
Nope, that's a thing needed for the automatic semicolon insertion where it fails. Without, JS parser will not be able to separate content of this line from content of the previous line. Maybe I was needed to put it at the end of the previous line, but if I remember correctly Standard JS has it like that at the beginning of the line. I will consult documentation tomorrow.

Reply:
oh that's interesting. I don't recall seeing it before

Reply:
Found it: https://standardjs.com/rules-en.html#semicolons. So the rule is: never start a line with `[` and if that happened to be the case, add `;` before. It really looks weird and unusual, but rule is the rule 🙂

Reply:
We won't have that often anyway. This is probably the first or the second time I'm using it